clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -669,7 +668,8 @@ public void onFailure(RuntimeException e) {
     private RequestFuture<Map<TopicPartition, OffsetData>> sendListOffsetRequest(final Node node,
                                                                                  final Map<TopicPartition, Long> timestampsToSearch,
                                                                                  boolean requireTimestamp) {
-        ListOffsetRequest.Builder builder = ListOffsetRequest.Builder.forConsumer(requireTimestamp)
+        ListOffsetRequest.Builder builder = ListOffsetRequest.Builder
+                .forConsumer(requireTimestamp, IsolationLevel.READ_UNCOMMITTED)
```

Comment:
Hmm, instead of using hard-coded READ_UNCOMMITTED, shouldn't we use the isolation_level in the Fetcher?